### PR TITLE
feat: implement the ui for migration to 10.0

### PR DESCRIPTION
**Pull request by elevatebart**
Closes [UNIFY-514](https://cypress-io.atlassian.net/browse/UNIFY-514)

### ScreenShots

These screenshots have been taken with the `cy.screenshot()` feature

MigrationWizard -- renders Automatic rename

MigrationWizard -- renders Config File migration

MigrationWizard -- renders Manual rename

### How to test

- Start component testing in the launchpad
- Move to the migration folder
- See the tests passing

### Todo next

in a separate PR, we should wire up the data to
- Detect that a migration is needed and suggest it (not force it)
- List the files detected by the original globs + compute the modified list
- Create mutation to actually apply Rename spec files

- Show contents of `cypress.json` & `cypress/plugins/index.js` and what it will look like when merged
- Make the mutation to apply the change: plugins + `cypress.json` -> `cypress.config.js`

### CODE REVIEW

**Comment by marktnoonan** [comment]
@elevatebart, I pulled this branch, ran `yarn` and then `yarn dev` inside of `packages/launchpad`. It opens the launchpad in the middle of the migration helper and logs a GraphQL error. I tried clicking "I'll do this later" but it had no effect.

The UI looks gorgeous though.

Error message
Launchpad stuck on migration helper

File: packages/graphql/src/schemaTypes/objectTypes/gql-Migration.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+export const MigrationStepEnum = enumType({
+  name: 'MigrationStepEnum',
+  members: MIGRATION_STEPS,
+})
+
+// TODO: implement these values for migration using the ctx
+export const Migration = objectType({
+  name: 'Migration',
+  description: 'The Wizard is a container for any state associated with initial onboarding to Cypress',
```
Comment (ZachJW34) [inline packages/graphql/src/schemaTypes/objectTypes/gql-Migration.ts:12]: Should be a description of the Migration objectType

File: packages/launchpad/src/migration/fragments/MigrationStep.vue [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  checked: false,
+})
+
+const slots = useSlots()
+const slotNames:string[] = []
+let i = 0
+
+while (slots[`step-${++i}`]) {
+  slotNames.push(`step-${i}`)
+}
```
Comment (ZachJW34) [inline packages/launchpad/src/migration/fragments/MigrationStep.vue:78]: Doesn't look like this is being used.

**Review by ZachJW34 — CHANGES_REQUESTED** [review]
UI looks great. I'm thinking some of the hover/click states should be reworked since alot of the items are disabled but they are styled to look like they are actionable.

Screen Shot 2021-12-21 at 11 25 52 AM
Screen Shot 2021-12-21 at 11 26 11 AM

Also, this feature blocks the normal flow so it's hard to get into the app without making src changes.

File: packages/graphql/src/schemaTypes/objectTypes/gql-Migration.ts (hunk 1 above)
Comment (elevatebart) [reply packages/graphql/src/schemaTypes/objectTypes/gql-Migration.ts:12]:
```suggestion
  description: 'The Migration contains will contain all data related to the 9.X to 10.0 migration UI',
```

**Comment by elevatebart** [comment]
@ZachJW34 The items will be actionable as soon as they become linked with the backend. 
The state is stored for now and navigating between different states means actions on the back-end. 

It might need some fine-tuning though because I do not know how to re-rename a file 😆

**Comment by ZachJW34** [comment]
@elevatebart makes sense. Looks like you've got some fun tests to fix, but once they are LGTM.

**Review by ZachJW34 — DISMISSED** [review]

**Review by ZachJW34 — APPROVED** [review]

**Pull request merged**